### PR TITLE
Unifying test arguments and test workflows

### LINKED ISSUE #393: Unify invoking the test workflows
The test workflows need restructuring for all the test suites. Today each test suite is using its own model of invoking the tests via test.sh or python <test_suite>.py.

Follow up of PR: https://github.com/opensearch-project/opensearch-build/pull/377

### PR DESCRIPTION
**Pull request by saratvemulapalli**
### Description
- Unifying test arguments across test suites and have a single runner via `test.sh`.
- Restructured BWC tests to pull manifests from S3 and kick off the test suite.
- Adding Jenkins job support for BWC tests
 
### Issues Resolved
Closes #393 #492 
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: bundle-workflow/README.md [hunk 1]
```diff
@@ -103,41 +103,52 @@ Tests the OpenSearch bundle.
...
 
-The following example kicks off all test suites for a distribution of OpenSearch 1.1.0.
-
+Usage:
 ```bash
-./bundle-workflow/test.sh manifests/opensearch-1.1.0.yml
+export AWS_ROLE_ARN=arn:aws:iam::<AWS_JENKINS_ACCOUNT>:role/opensearch-test
+export AWS_ROLE_SESSION_NAME=dummy-session
+
+Next, configure temporary credentials in environment w/
```
Comment (dblock) [inline bundle-workflow/README.md:111]: Quote only the shell commands, this line should not be between ```.

File: bundle-workflow/jenkins_workflow/test/Jenkinsfile [hunk 2]
```diff
@@ -34,6 +34,11 @@ pipeline {
                                             name: 'test_run_id',
                                             trim: true
                                     ),
+                                    string(
+                                            defaultValue: '',
+                                            name: 'test_suite_type',
```
Comment (dblock) [inline bundle-workflow/jenkins_workflow/test/Jenkinsfile:39]: Nit, `test_suite_type` -> `test_suite`, I think it's equally clear but shorter.

File: bundle-workflow/jenkins_workflow/test/README.md [hunk 3]
```diff
@@ -12,4 +12,4 @@ This job runs integration tests for a bundle via `test.sh` in Jenkins.
 | build_id |  Unique identifier for a bundle build                         |
 | architecture | CPU Architecture of bundle                                |
 | test_run_id | Unique identifier for a test run                           |
- 
\ No newline at end of file
+| test_suite_type | Run a specific test suite. [integ-test, bwc-test]      |
```
Comment (dblock) [inline bundle-workflow/jenkins_workflow/test/README.md:15]: test_suite

**Review by dblock — APPROVED** [review]
Some nits, but feel free to defer to another PR, there's enough here.

By making `bundle-workflow/src/run_integ_test.py` and friends runnable it becomes rather difficult to test them (hence lack of tests for them today). You could refactor them into classes such as `IntegTestSuiteRunner(TestSuite)`, sinking any common args logic into `TestSuite` and moving the runners into the test suite subfolders as well, and then calling `.run`. Then `test.sh` wouldn't need to do the switching of what test suite to run in bash and become a dumb caller for `run.sh` like the other runners.

File: bundle-workflow/jenkins_workflow/test/Jenkinsfile (hunk 2 above)
Comment (saratvemulapalli) [reply bundle-workflow/jenkins_workflow/test/Jenkinsfile:39]: Sure, I like it. I have a follow up PR, will clean it up there.

**Comment by saratvemulapalli** [comment]
> Some nits, but feel free to defer to another PR, there's enough here.
> 
> By making `bundle-workflow/src/run_integ_test.py` and friends runnable it becomes rather difficult to test them (hence lack of tests for them today). You could refactor them into classes such as `IntegTestSuiteRunner(TestSuite)`, sinking any common args logic into `TestSuite` and moving the runners into the test suite subfolders as well, and then calling `.run`. Then `test.sh` wouldn't need to do the switching of what test suite to run in bash and become a dumb caller for `run.sh` like the other runners.

Yeah absolutely, I didnt want to do major restructuring. Will take care of this in a follow up PR.

File: bundle-workflow/README.md [hunk 4]
```diff
@@ -103,41 +103,52 @@ Tests the OpenSearch bundle.
...
+export AWS_SESSION_TOKEN=<value>
+export AWS_ACCESS_KEY_ID=<value>
+export AWS_SECRET_ACCESS_KEY=<value>
```
Comment (gaiksaya) [inline bundle-workflow/README.md:114]: Do we need this if we are using role in line 108?

File: bundle-workflow/src/test_workflow/test_args.py [hunk 5]
```diff
@@ -12,16 +12,35 @@
...
 class TestArgs:
-    manifest: str
+    s3_bucket: str
+    opensearch_version: str
+    build_id: int
+    architecture: str
+    test_run_id: int
     component: str
     keep: bool
+    logging_level: int
```
Comment (gaiksaya) [inline bundle-workflow/src/test_workflow/test_args.py:22]: How is logging_level `int`?

File: bundle-workflow/README.md (hunk 4 above)
Comment (saratvemulapalli) [reply bundle-workflow/README.md:114]: Good question. To assume a role, it needs the right credentials. 
- For Jenkins, its taken care by the instance role.
- For developers on local desktops, they have to export these creds to assume that role.

File: bundle-workflow/src/test_workflow/test_args.py (hunk 5 above)
Comment (saratvemulapalli) [reply bundle-workflow/src/test_workflow/test_args.py:22]: Logging levels are mapped to integers internally by python.
Take a look at: https://docs.python.org/3/library/logging.html#logging.Logger.getEffectiveLevel

File: bundle-workflow/src/run_bwc_test.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+from system.temporary_directory import TemporaryDirectory
+from test_workflow.bwc_test.bwc_test_suite import BwcTestSuite
+from test_workflow.test_args import TestArgs
+
+
+def main():
+    args = TestArgs()
+    with TemporaryDirectory(keep=args.keep) as work_dir:
+        logging.info("Switching to temporary work_dir: " + work_dir)
+        os.chdir(work_dir)
```
Comment (owaiskazi19) [inline bundle-workflow/src/run_bwc_test.py:22]: Can use `with WorkingDirectory(work_dir) as cur_dir` here

Comment (saratvemulapalli) [reply bundle-workflow/src/run_bwc_test.py:22]: Sure will take care of this in a quick follow up PR.

**Review by owaiskazi19 — APPROVED** [review]

**Pull request merged**